pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPU.hpp

```diff
@@ -319,10 +319,9 @@ class AdjointJacobian final
             if (ops.hasParams(op_idx)) {
                 if (current_param_idx == *tp_it) {
                     const PrecisionT scalingFactor =
-                        BaseType::applyGenerator(
-                            mu, ops.getOpsName()[op_idx],
-                            ops.getOpsWires()[op_idx],
-                            !ops.getOpsInverses()[op_idx]) *
+                        this->applyGenerator(mu, ops.getOpsName()[op_idx],
```

Comment:
`this` isn't really used in C++ codebases for method resolution, unless you need the object pointer. Is the base-type not sufficient here? or just a general `applyGenerator` if defined locally.

Reply:
I'm unsure about this one, but while `BaseType` is defined a several places, it appears essentially only used in L-GPU. In the case of `adjointJacobian`, L-Qubit calls `mu`'s `applyGenerator` method directly or accesses other methods with `this` and L-Kokkos uses `this` throughout. Would you prefer we switch to `BaseType` in all classes?